lib/fuel-auth.js

```diff
@@ -11,18 +11,33 @@ const merge = require('lodash.merge');
 const version = require('../package.json').version;
 
 module.exports = class FuelAuth {
-	constructor(options) {		
+	constructor(options) {	
+		var isOauth2Flow = options && options.authOptions ? options.authOptions === 2 : false;
+
 		if (options) {
-			if (!options.clientId || !options.clientSecret) {
+			if (!options.clientId) {
 				throw new Error('clientId or clientSecret is missing or invalid');
 			}
-
-			if (typeof options.clientId !== 'string' || typeof options.clientSecret !== 'string') {
+			if (typeof options.clientId !== 'string') {
 				throw new Error('clientId or clientSecret must be strings');
 			}
-
-			if (options.authOptions && typeof options.authOptions.authVersion === 2 && !options.authUrl) {
-				throw new Error('Auth URL is mandatory for OAuth2 Authentication');
+			if(isOauth2Flow){
+				if(options.authOptions.applicationType !== 'public'){
+					if (!options.clientSecret) {
```

Comment:
The clientSecret is validated for value and type only for `isOauth2Flow === true`. The tests from the `general-tests.js` file do not catch this regression. It appears that mocha is happily reporting a test as passed even if no assert is executed.

For example, just comment the assert line in the catch block and run the test - it will pass! (which is wrong)
```
it('AuthorizationCode and RedirectURI are mandatory for public/web app', () => {
		let AuthClient;
		
		var options = {
			clientId:'client_id',
			clientSecret:'client_secret',
			authOptions:{
				authVersion:'2',
				applicationType: 'public'
			}
		};
		try {
			AuthClient = new FuelAuth(options);
		} catch (err) {
			//assert.equal(err.message, 'RedirectURI and Authorization Code are required for Public/Web App OAuth2 Authentication');
		}
	});
```

I recommend replacing all the test that use try and catch with `assert.throws(fn[, error][, message]) `https://nodejs.org/api/assert.html#assert_assert_throws_fn_error_message or adding an assert.fail("should throw") after the line that should throw the exception.

Reply:
updated